torchqdynamics/utils/td_tensor.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import get_args
+
+import torch
+from torch import Tensor
+
+from .tensor_types import OperatorLike, TDOperatorLike, cdtype, rdtype, to_tensor
+
+
+def to_td_tensor(
+    x: TDOperatorLike,
+    dtype: torch.dtype | None = None,
+    device: torch.device | None = None,
+    is_complex: bool = False,
+) -> TDTensor:
+    """Convert a `TDOperatorLike` object to a `TDTensor` object."""
+    if isinstance(x, get_args(OperatorLike)):
+        # convert to tensor
+        x = to_tensor(x, dtype=dtype, device=device, is_complex=is_complex)
+
+        return ConstantTDTensor(x)
+    elif callable(x):
+        # get default dtype and device
+        if dtype is None:
+            if is_complex:
+                dtype = cdtype(dtype)
+            else:
+                dtype = rdtype(dtype)
+        if device is None:
+            device = get_default_device()
```

Comment:
With `device = None` the device will be automatically set to the default device
```suggestion
```

Reply:
Yes but the goal is to feed it to the `check_callable` function which doesn't know about that. I can call it within the function if you'd prefer however.